compute_sdk/globus_compute_sdk/sdk/diagnostic.py

```diff
@@ -278,7 +278,39 @@ def general_commands_list(gc_base: Path):
     ]
 
 
-def get_diagnostic_commands(log_bytes: int, config_dir: str | None = None):
+def get_recent_endpoints(base_dir: Path, ep_names: list | None = None) -> list:
+    """
+    Get recently active endpoints based on latest file in endpoint
+    directory or the date of the directory itself.
+    The results are filtered by the given names, if provided
+
+    :param base_dir:   The base GC directory, normally ~/.globus_compute
+    :param ep_names:   List of endpoint names to filter by
+    :return:           List of endpoints in descending modified time (latest first)
+    """
+    ep_list_by_ts = []
+    if ep_names:
+        ep_list = [base_dir / n for n in ep_names]
+    else:
+        ep_list = [p for p in base_dir.iterdir()]
+        ep_names = [p.name for p in base_dir.iterdir()]
+    for ep in ep_list:
+        if ep.is_dir() and ep.name in ep_names:
+            latest_ts = ep.stat().st_mtime
+
+            for ep_item in ep.iterdir():
+                if ep_item.stat().st_mtime > latest_ts:
+                    latest_ts = ep_item.stat().st_mtime
```

Comment:
Minor bikeshed, but one that prefers the use of `max()` as a descriptor of what the loop intention is:
```python
latest_ts = max(ep.stat().st_mtime, *(p.stat().st_mtime for p in ep.iterdir()))
```